WalletWasabi/WabiSabi/Client/UtxoSelectionParameters.cs

```diff
@@ -0,0 +1,58 @@
+using NBitcoin;
+using System.Collections.Immutable;
+using System.Linq;
+using WabiSabi.Crypto.Randomness;
+using WalletWasabi.Extensions;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+public record UtxoSelectionParameters(
+	MoneyRange AllowedInputAmounts,
+	Money MinAllowedOutputAmount,
+	CoordinationFeeRate CoordinationFeeRate,
+	FeeRate MiningFeeRate,
+	ImmutableSortedSet<ScriptType> AllowedInputScriptTypes)
+{
+	// for testing only
+	public static UtxoSelectionParameters FromRoundParameters(RoundParameters roundParameters) =>
+		FromRoundParameters(roundParameters, [ScriptType.P2WPKH, ScriptType.Taproot]);
+
+	public static UtxoSelectionParameters FromRoundParameters(RoundParameters roundParameters, ScriptType[] scriptTypesSupportedByWallet)
+	{
+		var outputTypes = roundParameters.AllowedOutputTypes.Intersect(scriptTypesSupportedByWallet);
+		var maxVsizeInputOutputPairScriptType = outputTypes.MaxBy(x => x.EstimateInputVsize() + x.EstimateOutputVsize());
+		var smallestEffectiveDenom = DenominationBuilder.CreateDenominations(
+				roundParameters.CalculateMinReasonableOutputAmount(scriptTypesSupportedByWallet),
+				roundParameters.AllowedOutputAmounts.Max,
+				roundParameters.MiningFeeRate,
+				[maxVsizeInputOutputPairScriptType],
+				new InsecureRandom()) // Random generator is not used and then the algorithm is deterministic
```

Comment:
The RNG is used to pick randomly one of the available script type. In this case there is exactly one script type available and then the random number is useless. It was passed all the way down just because the method receives a RNG but it is unnecessary. In fact, if the RNG were used then this method would not be deterministic and that would be a problem.